tcms/issuetracker/tests/test_azure.py

```diff
@@ -0,0 +1,199 @@
+# pylint: disable=attribute-defined-outside-init
+
+import os
+import re
+import time
+import unittest
+
+from django.utils import timezone
+
+from tcms.core.contrib.linkreference.models import LinkReference
+from tcms.issuetracker.types import AzureDevops
+from tcms.issuetracker.azure_integration import AzureAPI
+from tcms.rpc.tests.utils import APITestCase
+from tcms.testcases.models import BugSystem
+
+from tcms.tests.factories import ComponentFactory
+from tcms.tests.factories import TestExecutionFactory
+
+
+@unittest.skipUnless(os.getenv('TEST_BUGTRACKER_INTEGRATION'),
+                     'Bug tracker integration testing not enabled')
+class TestAzureIntegration(APITestCase):
+    existing_bug_id = 1
+    # TODO: Replace with a Kiwi specific url for test.
+    existing_bug_url = 'https://dev.azure.com/cmetebahadir/KiwiAzureTCMS/_workitems/edit/33'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.execution_1 = TestExecutionFactory()
+        self.execution_1.case.summary = "Tested at " + timezone.now().isoformat()
+        self.execution_1.case.text = "Given-When-Then"
+        self.execution_1.case.save()  # will generate history object
+        self.execution_1.run.summary = "Automated TR for Azure integration on " + \
+                                       timezone.now().isoformat()
+        self.execution_1.run.save()
+
+        self.component = ComponentFactory(name='GitHub integration',
+                                          product=self.execution_1.run.plan.product)
+        self.execution_1.case.add_component(self.component)
+
+        bug_system = BugSystem.objects.create(  # nosec:B106:hardcoded_password_funcarg
+            name='Azure for kiwitcms/test-azure-integration',
+            tracker_type='tcms.issuetracker.types.AzureDevops',
+            base_url='https://dev.azure.com/cmetebahadir/KiwiAzureTCMS',
+            # TODO: Replace with an env variable like "AZURE_BUGTRACKER_INTEGRATION_API_TOKEN"
+            # api_password=os.getenv('AZURE_BUGTRACKER_INTEGRATION_API_TOKEN'),
+            api_password='nufqr2cg3forutzkdnskqyjc5qw5mi6jalmbpiud45h264a6mmja',
```

Comment:
Don’t get mad already revoked ☺️, it is hardcoded only  for initial tests.

Reply:
Regardsless it is bad practice. Use os.getenv and export the appropritate values locally before running the tests.

Reply:
On a related note how do I get an API token so I can configure that in the repo secrets ?

Reply:
Next to you user avatar on top right corner after logged in to https://dev.azure.com/, there is a settings drop-down its on below.

![image](https://user-images.githubusercontent.com/8937774/97195383-527ac480-17bc-11eb-9f75-d8aa863d771b.png)

It should have these rights to modify work items.

![image](https://user-images.githubusercontent.com/8937774/97195494-776f3780-17bc-11eb-9da3-20b2b6d7b3dd.png)

Reply:
Thanks. I've granted **Read & write** permission, not sure what **manage** is for. This is available as `AZURE_BOARDS_INTEGRATION_API_TOKEN`